### PR TITLE
Add Dockerfile

### PR DESCRIPTION
**Pull request by mbyczkowski**
This should be pretty straightforward. The only trick is that I used alpine docker image for the base, so I had to include tzinfo-data gem to get rails to work properly.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,28 @@
+FROM mhart/alpine-node:5.0
+
+ENV BUILD_PACKAGES bash curl-dev ruby-dev build-base libffi-dev libpq postgresql-dev git
+ENV RUBY_PACKAGES ruby ruby-io-console ruby-bundler ruby-raindrops ruby-nokogiri
```
Comment (jeromegn) [inline Dockerfile]: You probably don't need libffi, libpq, postgresql-dev and maybe not even git :)

Raindrops is for the unicorn gem, which I don't think we use in this project.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    apk upgrade && \
+    apk add $BUILD_PACKAGES && \
+    apk add $RUBY_PACKAGES && \
+    rm -rf /var/cache/apk/*
+
+RUN mkdir /app
+WORKDIR /app
+
+ADD . /app
+RUN bundle install --jobs 4 --without development test
```
Comment (jeromegn) [inline Dockerfile]: The main difference of not copying the `Gemfile` and `Gemfile.lock` before (I guess in this case, you'd also have to copy the `gemspec` file), is that the build will have to bundle all gems every time there's any change in the `app/`'s files.

If we just `COPY` the files bundle needs to know about first, then the bundling phase will only happen if there are changes in the file. Else, it'll use the cached layer which is all the gems installed.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+RUN npm install
+
+WORKDIR /app
+
+ENV RAILS_ENV production
+RUN cd server && bundle exec rake tmp:cache:clear
+RUN cd server && bundle exec rake assets:precompile
```
Comment (jeromegn) [inline Dockerfile]: Just a note: The documentation app is really simple, it behaves like a normal Rails app including the gem. That means, it pulls the assets from our Cloudfront CDN.

However, it does have some assets, and so precompiling them will make the app faster for the first load :)

File: Dockerfile (hunk 1 above)
Comment (mbyczkowski) [inline Dockerfile]: cool, I will try to simplify that :+1:

File: Dockerfile (hunk 2 above)
Comment (mbyczkowski) [inline Dockerfile]: That's a good point! I was reading a bit on Dockerfile later and that's what I got out of it. The progrem I had with megatron (after copying `gemspec` file, too), was that it included more stuff as a reference (so things in `lib/` dir). I can try again with only adding `gemspec` and `lib` dir and see what happens.

Comment (jeromegn) [inline Dockerfile]: Ah, that's interesting. It does indeed load a lot of things...

It's not a big deal, it's probably just a matter of a minute extra in the build process, and this app is not critical.

**Comment by jeromegn** [comment]
lgtm

**Pull request merged**